docs/StardustDocs/topics/readSqlFromCustomDatabase.md

```diff
@@ -0,0 +1,167 @@
+[//]: # (title: How to Extend DataFrame Library for Custom SQL Database Support: Example with HSQLDB)
+
+# How to Extend DataFrame Library for Custom SQL Database Support: Example with HSQLDB
+
+This guide demonstrates how advanced users can extend the Kotlin DataFrame library to support a custom SQL database, using HSQLDB as an example. By following these steps, you will be able to integrate your custom database into the DataFrame library, allowing for seamless DataFrame creation, manipulation, and querying.
+
+---
+
+## Prerequisites
+
+1. **Create a Gradle Project**:  
+
+Add the following dependencies to your `build.gradle.kts`:
+
+```kotlin
+dependencies {
+    implementation("org.jetbrains.kotlinx:dataframe:$dataframe_version")
+    implementation("org.hsqldb:hsqldb:$version")
+}
+```
+
+2. **Install HSQLDB**:
+
+Follow the [HSQLDB Quick Guide](https://www.tutorialspoint.com/hsqldb/hsqldb_quick_guide.htm) to set up HSQLDB locally.
+
+3. **Start the HSQLDB Server**:
+
+Launch a terminal or command prompt and execute the following command:
+
+```bash
+java -classpath lib/hsqldb.jar org.hsqldb.server.Server --database.0 file:hsqldb/demodb --dbname.0 testdb
+```
+
+## Implementing Custom Database Type Support
+
+To enable HSQLDB integration, implement a custom `DbType` by overriding required methods.
+
+
+**Create the HSQLDB Type**
+
+```kotlin
+import org.jetbrains.kotlinx.dataframe.io.TableColumnMetadata
+import org.jetbrains.kotlinx.dataframe.io.TableMetadata
+import org.jetbrains.kotlinx.dataframe.io.db.DbType
+import org.jetbrains.kotlinx.dataframe.schema.ColumnSchema
+import java.sql.ResultSet
+import java.util.Locale
+import kotlin.reflect.KType
+
+/**
+ * Represents the HSQLDB database type.
+ *
+ * This class provides methods to convert data from a ResultSet to the appropriate type for HSQLDB,
```

Comment:
methods is the java name, I think Kotlin calls it "functions"

Reply:
In official Kotlin documentations, method is very popular https://kotlinlang.org/docs/idioms.html#call-multiple-methods-on-an-object-instance-with